src/shared/components/copyDownloadControls/SimpleCopyDownloadControls.tsx

```diff
@@ -94,6 +95,16 @@ export class SimpleCopyDownloadControls extends React.Component<
         }
     }
 
+    /**
+     * Wrapper around downloadData() to return as a Promise<string> for ICopyDownloadButtonsProps
+     * See TECH_DOWNLOADDATA
+     */
+    private downloadDataAsPromise(): Promise<string> | undefined {
+        if (this.props.downloadData) {
+            return Promise.resolve(this.props.downloadData());
+        }
+    }
+
```

Comment:
See comment in CopyDownloadControls.tsx about why downloadData is a Promise<string>. Here we have a wrapper to convert the synchronous case to a Promise<string>.